src/ProtocolFees.sol

```diff
@@ -50,7 +50,7 @@ abstract contract ProtocolFees is IProtocolFees, PausableRole {
             if (gasleft() < controllerGasLimit) revert ProtocolFeeCannotBeFetched();
 
             (bool _success, bytes memory _data) = address(protocolFeeController).call{gas: controllerGasLimit}(
-                abi.encodeWithSelector(IProtocolFeeController.protocolFeeForPool.selector, key)
+                abi.encodeCall(IProtocolFeeController.protocolFeeForPool, (key))
             );
```

Comment:
tbh i never used `encodeCall` before 😂.. Just in case someone is with me 

![CleanShot 2024-05-31 at 10 09 42@2x](https://github.com/pancakeswap/pancake-v4-core/assets/137024020/b01e65d2-d6ce-476c-9064-a0f6c9acb4fc)

https://soliditylang.org/blog/2021/12/20/solidity-0.8.11-release-announcement/